src/include/duckdb/execution/index/art/node.hpp

```diff
@@ -9,42 +9,61 @@
 #pragma once
 
 #include "duckdb/execution/index/art/art_key.hpp"
+#include "duckdb/execution/index/art/prefix.hpp"
 #include "duckdb/common/common.hpp"
 #include "duckdb/storage/meta_block_writer.hpp"
 #include "duckdb/storage/meta_block_reader.hpp"
 #include "duckdb/storage/index.hpp"
-#include "duckdb/execution/index/art/prefix.hpp"
 
 namespace duckdb {
-enum class NodeType : uint8_t { N4 = 0, N16 = 1, N48 = 2, N256 = 3, NLeaf = 4 };
-
+enum class NodeType : uint8_t { NLeaf = 0, N4 = 1, N16 = 2, N48 = 3, N256 = 4 };
 class ART;
 class Node;
 class SwizzleablePointer;
 
 struct InternalType {
 	explicit InternalType(Node *n);
+
 	void Set(uint8_t *key_p, uint16_t key_size_p, SwizzleablePointer *children_p, uint16_t children_size_p);
+
 	uint8_t *key;
 	uint16_t key_size;
 	SwizzleablePointer *children;
 	uint16_t children_size;
 };
 
+struct MergeInfo {
+	MergeInfo(ART *l_art, ART *r_art, Node *&l_node, Node *&r_node)
+	    : l_art(l_art), r_art(r_art), l_node(l_node), r_node(r_node) {};
+	ART *l_art;
+	ART *r_art;
+	Node *&l_node;
+	Node *&r_node;
+};
+
+struct ParentsOfNodes {
+	ParentsOfNodes(Node *&l_parent, idx_t l_pos, Node *&r_parent, idx_t r_pos)
+	    : l_parent(l_parent), l_pos(l_pos), r_parent(r_parent), r_pos(r_pos) {};
+	Node *&l_parent;
+	idx_t l_pos;
+	Node *&r_parent;
+	idx_t r_pos;
+};
```

Comment:
Are the names of these structs clear? I had some trouble finding matching/descriptive names.